### PR TITLE
Refactor atomic habits feature

### PR DESCRIPTION
**Pull request by haoyangw**
`Command` and `Manager` superclasses are now available, which features are expected to utilise.

Atomic habits doesn't yet do this, and defines its own abstract classes.

Refactor atomic habits classes to make use of the new superclasses, and update `MainManager` to support atomic habit commands now(via `AtomicHabitManager`).

@YongbinWang @BernardLesley Require your review to ensure that no regressions were introduced, and code function as you intended initially. Thank you all for writing the feature, by the way!

This addresses #60

### CODE REVIEW

**Review by YongbinWang — COMMENTED** [review]
Overall very well integrated code with the superclass nicholas created. Seems to be functional, some minor issues to be discussed.

File: src/main/java/wellnus/ui/TextUi.java [hunk 1]
```diff
@@ -22,9 +23,17 @@ public class TextUi {
             + "please key in your input!!";
     private static final String BUFFER_OVERFLOW_MSG = "Your input is too long,"
             + "please shorten it!!";
-    private static final Scanner SCANNER = new Scanner(System.in);
+    private final Scanner scanner;
     private String separator = "-";
 
+    public TextUi() {
+        this(System.in);
+    }
```
Comment (wenxin-c) [inline src/main/java/wellnus/ui/TextUi.java:31]: this constructor will not create a scanner object i think

File: src/test/java/wellnus/atomichabit/AtomicHabitTest.java [hunk 2]
```diff
@@ -1,33 +1,76 @@
...
         String payload = "junit test";
         String expectedOutput = "Yay! You have added a new habit:"
                 + System.lineSeparator()
                 + "'"
                 + payload
                 + "'"
                 + " was successfully added";
-        Command command = new AddCommand(payload);
-        Assertions.assertEquals(expectedOutput, command.execute(habitList).getCommandResult());
+        String testCommand = "hb add --name " + payload;
```
Comment (wenxin-c) [inline src/test/java/wellnus/atomichabit/AtomicHabitTest.java:69]: May I check are we using feature name in command? iirc we said we gonna remove all the feature names in commands

**Review by nichyjt — APPROVED** [review]
LGTM! All the comments have been addressed.

**Pull request merged**